components/sysutils/nut/Makefile

```diff
@@ -101,26 +168,97 @@ CONFIGURE_OPTIONS +=	--with-net-snmp-config=$(USRBINDIR)/net-snmp-config-$(BITS)
 
 CONFIGURE_OPTIONS +=	$(CONFIGURE_OPTIONS.$(BITS))
 
+# Fiddle with packaging:
+# * move man1m pages to target directory (see also patches/07-man.patch)
+# * move python stuff to vendor-packages
+# * make sure full python program name is in the shebang
+#   (and note we can not use ...\1${PYVER} because it is
+#    also numbers and SED gets confused about group \12 or so)
+# * pre-compile PYC files
+# * apply a little hack to comment away the import which
+#   confuses pkgdepend (it looks for "import" lines and
+#   disregards "try/except" around it)
+COMPONENT_POST_INSTALL_ACTION += \
+    $(MKDIR) $(PROTOUSRSHAREMANDIR)/man1m \
+        && $(MV) $(PROTOUSRSHAREMANDIR)/man1/*.1m $(PROTOUSRSHAREMANDIR)/man1m/ \
+        || { echo "FAILED to fix up man1m" >&2 ; exit 1; } ; \
+    for PYVER in $(PYTHON_VERSIONS.NUTUI) ; do \
+        $(MKDIR) "$(PROTOUSRLIBDIR)/python$${PYVER}/vendor-packages" \
+            && $(MV) "$(PROTOUSRLIBDIR)/python$${PYVER}/site-packages/"* "$(PROTOUSRLIBDIR)/python$${PYVER}/vendor-packages/" \
+            && find "$(PROTOUSRLIBDIR)/python$${PYVER}/vendor-packages/" -name '*.py' | while read F ; do \
+                $(GSED) '/^\(\#\!.*\/python\)$$/ s/$$/'"$${PYVER}"'/' -i "$$F" || exit ; \
+            done \
+        && python$${PYVER} -m compileall "$(PROTOUSRLIBDIR)/python$${PYVER}/vendor-packages" || exit ; \
+    done; \
+    if [ -n "$(PYVER.2X)" ] ; then \
+        python$(PYVER.2X) -c "import pygtk;pygtk.require('2.0');import pynotify" >/dev/null || { \
+            echo "Neutering import of pynotify in NUT-Monitor-py2gtk2 for this build" >&2 ; \
+            $(GSED) -i "$(PROTOUSRSHAREDIR)/nut/nut-monitor/app/NUT-Monitor-py2gtk2" \
+                -e 's,\(\s\)\(import pynotify\),\1return; \#\#\#NOT_HERE_\2,' || exit ; \
+        } ; \
+    fi;
+
+test: $(TEST_64)
+
 # Manually added dependencies:
 REQUIRED_PACKAGES += text/asciidoc
 REQUIRED_PACKAGES += library/augeas-tools
+#REQUIRED_PACKAGES += system/library/usb/libusb
+REQUIRED_PACKAGES += library/libusb-1
+#REQUIRED_PACKAGES += library/security/openssl
+REQUIRED_PACKAGES += library/security/openssl-31
+
+$(foreach PYVAR,$(PYTHON_VERSIONS.NUTUI),\
+    $(eval REQUIRED_PACKAGES += runtime/python-$(subst .,,$(PYVAR))))
+
+###REQUIRED_PACKAGES += library/python/pygtk2
+REQUIRED_PACKAGES += library/python/pygtk2-$(PYV.2X)
+REQUIRED_PACKAGES += library/python/pygobject-$(PYV.2X)
+###REQUIRED_PACKAGES += library/python/pygobject-3-$(PYV.2X)
+REQUIRED_PACKAGES += library/python/pyqt5
+
+# Obsoleted since 2.7 (packaging at least)
+ifneq (,$(PYV.2X))
+ifneq (27,$(PYV.2X))
+REQUIRED_PACKAGES += library/python-2/python-notify-$(PYV.2X)
+endif
+endif
+#REQUIRED_PACKAGES += library/python/notify2
 
 # Auto-generated dependencies
 REQUIRED_PACKAGES += $(GCC_RUNTIME_PKG)
 REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
 REQUIRED_PACKAGES += library/gd
 REQUIRED_PACKAGES += library/libtool/libltdl
+REQUIRED_PACKAGES += library/libmodbus
 REQUIRED_PACKAGES += library/neon
```

Comment:
Did you really run gmake REQUIRED_PACKAGES? If I do it here, I'll get some differences:
 # Auto-generated dependencies
+PYTHON_REQUIRED_PACKAGES += library/python/pygobject
+PYTHON_REQUIRED_PACKAGES += library/python/pygtk2
+PYTHON_REQUIRED_PACKAGES += runtime/python
 REQUIRED_PACKAGES += $(GCC_RUNTIME_PKG)
 REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
 REQUIRED_PACKAGES += library/gd
-REQUIRED_PACKAGES += library/libtool/libltdl
 REQUIRED_PACKAGES += library/libmodbus
+REQUIRED_PACKAGES += library/libtool/libltdl
+REQUIRED_PACKAGES += library/libusb-1
 REQUIRED_PACKAGES += library/neon
+REQUIRED_PACKAGES += library/python/pyqt5
+REQUIRED_PACKAGES += library/security/openssl-31
 REQUIRED_PACKAGES += shell/ksh93
-REQUIRED_PACKAGES += SUNWcs
 REQUIRED_PACKAGES += system/library
 REQUIRED_PACKAGES += system/library/math
 REQUIRED_PACKAGES += system/management/freeipmi
-#REQUIRED_PACKAGES += system/management/nut/libs
 REQUIRED_PACKAGES += system/management/powerman
 REQUIRED_PACKAGES += system/management/snmp/net-snmp

You should avoid editing behind the #Auto-generated dependencies marker. It will be changed by the next gmake REQUIRED_PACKAGES run. If you need to manually add packages then use the area above.

Reply:
Did at some point, then it was failing (some chicken-and-egg, without proper stuff in the Makefile it could not complete processing of p5m files), so possibly missed on the last loop. Will recheck.

Reply:
Thinking of it, possibly it was not run as a standalone rule (lately at least) but as part of `gmake -j 20 clean publish sample-manifest REQUIRED_PACKAGES` sort of incantation. Is there something that might require it to be a standalone step?

Reply:
Revised, some manually added stuff above dropped.

Reply:
Huh, the smart magic misfired somewhat:
````
pkg list: no packages matching the following patterns are installed:
  /library/python/pygobject-39
  /library/python/pygtk2-39
Adding required packages to build environment...
````

These are only needed for the 2.x version of UI; the 3.x one is with QT5.

Reply:
Curiously, it left `REQUIRED_PACKAGES += library/python/pyqt5` alone, without magic auto-versioning.

Reply:
> Thinking of it, possibly it was not run as a standalone rule (lately at least) but as part of `gmake -j 20 clean publish sample-manifest REQUIRED_PACKAGES` sort of incantation. Is there something that might require it to be a standalone step?

Please do not use `-j` when invoking `gmake`.  Instead, you can use something like `gmake PARALLEL_JOBS=20 ...` if default `8` is not enough for you.

Yes, you should make sure that `publish`, `sample-manifest`, `REQUIRED_PACKAGES` (and also `test`) targets pass and no files got changed in your component directory after they are run.  If all these targets pass and no files are changed then their order is not so important, but if that's not the case (usually during the component integration development) then the recommended order is this (before going to next step the current step should pass; you can return to any previous step if needed):

1. First make sure `sample-manifest` produces what you want (edit `Makefile` and/or other files then `gmake clobber` then `env - gmake sample-manifest`).
2. `env - gmake test`.
3. Create/update your `*.p5m` file(s) based on `sample-manifest.p5m`.
4. `env - gmake REQUIRED_PACKAGES`.
5. `gmake clobber` then `env - gmake publish`.